### PR TITLE
Consolidating boskos scripts for CAPA

### PR DESCRIPTION
**Pull request by SanikaGawhane**
**What this PR does / why we need it**:

- Consolidates scripts `hack/checkin_account.py`, `hack/checkout_account.py` and `hack/heartbeat_account.py` into one concise `hack/boskos.py` and eliminates duplication of functions performed repetitively in separate scripts. 

- Image builder has a [PR](https://github.com/kubernetes-sigs/image-builder/pull/423) open for similar changes for AMIs and this PR can ensure uniformity in upstream CAPA.

### CODE REVIEW

**Comment by randomvariable** [comment]
/hold

for ci results.

File: scripts/ci-conformance.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ if [ -n "${BOSKOS_HOST:-}" ]; then
   # Check out the account from Boskos and store the produced environment
   # variables in a temporary file.
   account_env_var_file="$(mktemp)"
-  python hack/checkout_account.py 1>"${account_env_var_file}"
+  python hack/boskos.py --get 1>"${account_env_var_file}"
```
Comment (kkeshavamurthy) [inline scripts/ci-conformance.sh:44]: Need to use python3 to call the boskos script everywhere.

**Comment by randomvariable** [comment]
We need the license headers at the top of files please.

**Comment by randomvariable** [comment]
`#!/usr/bin/env python`
needs to be updated to `#!/usr/bin/env python3` in https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/hack/boilerplate/boilerplate.py

**Comment by randomvariable** [comment]
@SanikaGawhane are you updating this PR?

**Comment by SanikaGawhane** [comment]
@randomvariable Yes, this PR is still WIP.

**Comment by richardcase** [comment]
> Yes, this PR is still WIP.

Thanks for working on this @SanikaGawhane. Would you be able to add the `WIP:` prefix to the title of the PR whilst its WIP?

File: hack/boskos.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+
+import requests
```
Comment (kkeshavamurthy) [inline hack/boskos.py:21]: `requests` module is not installed on the kubekins image.

File: scripts/ci-e2e.sh [hunk 3]
```diff
@@ -83,7 +86,7 @@ make test-e2e ARTIFACTS=$ARTIFACTS
 test_status="${?}"
 
 # If Boskos is being used then release the AWS account back to Boskos.
-[ -z "${BOSKOS_HOST:-}" ] || hack/checkin_account.py
+[ -z "${BOSKOS_HOST:-}" ] || hack/boskos.py --release
```
Comment (kkeshavamurthy) [inline scripts/ci-e2e.sh:89]: This needs to be something like 

`[ -z "${BOSKOS_HOST:-}" ] || python3 -u hack/boskos.py --release`

**Comment by randomvariable** [comment]
lgtm

looks like we can take off the wip?

**Pull request merged**